x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -214,8 +214,7 @@ private void buildUser(X509AuthenticationToken token, String principal, ActionLi
                 "pki_delegated_by_user",
                 token.getDelegateeAuthentication().getEffectiveSubject().getUser().principal(),
                 "pki_delegated_by_realm",
-                // TODO: this should be the realm of effective subject
-                token.getDelegateeAuthentication().getAuthenticatingSubject().getRealm().getName()
+                token.getDelegateeAuthentication().getEffectiveSubject().getRealm().getName()
```

Comment:
Mostly checking my own understanding here: `pki_delegated_by_realm` might be used in role mappings which might in turn break if we change the realm here. Is this the case?

Since this is strictly a bug fix, it's fine to make a non-BWC change without deprecation and involving the committee etc. I'm wondering though if we should add a `breaking:` note to the changelog. WDYT?

Reply:
This is a good topic to discuss. Thanks for raising it.

I checked briefly on the past PRs that have both `>bug` and `>breaking` and they are pretty rare. I think the breakage should be quite visible to users and users might intentional rely on the bug for the `>breaking` label to apply. The one PR with both `>bug` and `>breaking` that I raised in the past is to always require a name to be specified when creating API keys (#59836). This one had `>breaking` because we knew solution teams were relying on this behaviour (alerting was granting API keys without names).

For this particular case, I don't think the "breakage" will really noticeable. The PKI delegation is designed to be used by the Kibana system user so that end-user can authenticate with the PKI realm when Kibana is terminating TLS. For this breakage to manifest, the user's environment must have the kibana system user run-as a different user, then call the DelegatePKI API. In addition, there must a role mapping that relies on the kibana system user's realm which is either `reserved` or `_service_account`. Such a configuration would be really exotic if realistic at all. 

Therefore, I'd prefer to go without the `>breaking` label. Since the chance for anyone running into it is so minuscule, I'd avoid unnecessary concerns for users who read the breaking release notes and think it might apply to them.

Reply:
Excellent, I'm on board with skipping the `>breaking` label given that any impact is extremely unlikely. Thanks for giving more context on this.